web/source/kmwembedded.ts

```diff
@@ -13,27 +13,35 @@
 /*****************************************/
 
 namespace com.keyman.osk {
-  VisualKeyboard.prototype.optionKey = function(this: VisualKeyboard, e: KeyElement, keyName: string, keyDown: boolean) {
+
+   VisualKeyboard.prototype.optionKey = function(this: VisualKeyboard, e: KeyElement, keyName: string, keyDown: boolean) {
     let keyman = com.keyman.singleton;
+    let device = com.keyman.singleton.util.device;
 
     if(keyName.indexOf('K_LOPT') >= 0) {
-      if(keyDown) {
-        this.menuEvent = e;
-        if(typeof keyman['showKeyboardList'] == 'function') {
-          keyman['showKeyboardList']();
+      if (device.OS == 'Android') {
+        if (typeof keyman['doGlobeKey'] == 'function') {
+          keyman['doGlobeKey'](keyDown);
         }
       } else {
-        if(this.menuEvent) {
-          this.highlightKey(this.menuEvent, false);
-        }
-        if(typeof(window['menuKeyUp']) == 'function') {
-          window['menuKeyUp']();
-        }
-        this.menuEvent = null;
+        if(keyDown) {
+          this.menuEvent = e;
+          if(typeof keyman['showKeyboardList'] == 'function') {
+            keyman['showKeyboardList'];
+          }
+        } else {
+          if(this.menuEvent) {
+            this.highlightKey(this.menuEvent, false);
+          }
+          if(typeof(window['menuKeyUp']) == 'function') {
+            window['menuKeyUp']();
+          }
+          this.menuEvent = null;
+        } 
```

Comment:
Just going to point out at this point, `keyman['doGlobeKey'](true)` is called where `keyman['showKeyboardList']()` was, while `keyman['doGlobeKey'](false)` is called where `keyman['menuKeyUp']()` was.

You could very easily have Android's keyboard.html make the original two calls simple wrappers for your `doGlobeKey` without losing _anything_.

Reply:
ah, that works out nice. thanks for the tip!